src/panels/beaconSummaryPanel/BeaconSummaryPanel.tsx

```diff
@@ -42,9 +43,12 @@ export const BeaconSummaryPanel: FunctionComponent<IBeaconSummaryProps> = ({
     fetchBeacon(beaconId);
   }, [userState, beaconId, beaconsGateway]);
 
-  const handleSave = async (beacon: IBeacon): Promise<void> => {
+  const handleSave = async (updatedBeacon: Partial<IBeacon>): Promise<void> => {
     try {
-      await beaconsGateway.updateBeacon(beacon.id, beacon);
+      await beaconsGateway.updateBeacon(
+        beacon.id,
+        diffObjValues(beacon, updatedBeacon)
+      );
```

Comment:
I chose to place the responsibility for only submitting updated values in the `Panel` because:
- The `Gateway` should only be responsible for making calls, not manipulating data;
- The `Mapper` should only be responsible for translating between data formats, not making calls;

This left me with either the `EditingState` or the `Panel` components.
- Putting the `diffObjValues` in the `EditingState` felt iffy for me because it interferes with the basic pattern of all form values being submitted when the form is submitted.
- The `Panel` has access to the original beacon, the updated beacon and the gateway.  It felt natural for the `Panel` to receive the form submission, filter out unchanged fields and then call the relevant method on the gateway.

Grateful for thoughts!  Is there another design I haven't thought of?